docs/proposals/ha-configuration.md

```diff
@@ -0,0 +1,186 @@
+            <interface> = <string> - The interface to use.
+                                     Default: Default interface on node or eth0
+            <port> = <number> - Port to watch for resource availability.
+                                Default: 80.
+            <string> = a string of characters.
+            <number> = a number ([0-9]*).
+
+
+## Examples
+Examples:
+
+       $ # View the HA configuration.
+       $ openshift admin ha-config -o yaml
+
+       $ # Create HA configuration with IP failover serving 5 virtual IPS.
+       $ openshift admin ha-config --virtual-ips="10.1.1.5-8,42.42.42.42" \
+                                   --selector="jack=the-vipper"           \
+                                   --create
+
+       $ # Create a HA configuration with IP failover but disabling the
+       $ # VRRP multicast group (using unicast instead).
+       $ # Note: Initial release **will** likely not have unicast support.
+       $ openshift admin ha-config ha-amzn --unicast=true               \
+                                   --selector="ha-router=amzn-us-west"  \
+                                   --virtual-ips="54.192.0.42-43"       \
+                                   --watch-port=80  --create
+
+       $ # Delete a previously created HA configuration.
+       $ openshift admin ha-config ha-amzn --delete
+
+
+## Under-the-hood
+Under the hood, the HA configuration creates and starts up an HA sidecar
+pod on all the nodes matching the given selector - can only run one HA
+sidecar pod per node and the number of replicas is the number of nodes
+that match. It also starts watching for any changes to nodes
+(addition/deletion/modifications) and processes those change sets,
+adjusting the replica count in the deployment to match the reality.
+
+And from there on, kubernetes handles the task of ensuring there are
+```n``` replicas of the HA sidecar pod.
+
+On the nodes running the HA sidecar pod, keepalived ensures the watched
+service is available and ensures that the set of VIPs is available across
+the selection of nodes (or sub-cluster). This means an individual node
+would be a candidate to host zero or more VIPs if the watched service
+(example HAProxy on port 80) is available and 0 VIPs if the watched service
+is **not** available. This also allows handling the case where the number
+of VIPs is much smaller than the node selection or even nodes running the
+watched service (e.g. HAProxy router) - not all nodes would be allocated a
+virtual IP but would still be candidates on a failure.
+And the case where the number of VIPs is more than the node selection, a
+node could service multiple VIPs.
+
+This allows us to the cover:
+  1. The normal steady state case.
+  1. When a cluster is resized - nodes shrink or grow.
+  1. When a cluster is modified - really node labels are modified.
+  1. Failure cases when a node or watched service or network fails.
```

Comment:
Add a note here that the PerNodeController will remove the need to fiddle with this, because you will be able to run these on a set of nodes directly.